internal/pipeline/types_test.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package pipeline_test
+
+import (
+	"testing"
+
+	. "github.com/GoogleContainerTools/kpt/internal/pipeline"
+	"github.com/stretchr/testify/assert"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+func parseFromString(input string) (Pipeline, error) {
```

Comment:
I think defining constructors (or helpers) in the pipeline pkg itself will help to avoid this extra code in test pkg. To begin with, we can define following methods:

pipeline.FromFile(path string) (*Pipeline, error)

pipeline.FromReader(r io.Reader) (*Pipeline, error)

actually FromFile can use the FromReader()..

Reply:
I agree but this PR is intended to define types but not read/write API.

Reply:
hmm... I see. A good practice will be to test functionality (public methods) provided by a package. In this case, it makes it harder to see what exactly is it testing ? (serialization/deserialization ?). 

May be, we can just define the types in this PR. And include these tests with pipeline IO PR. That way, we know what exactly is being tested. WDYT ?

Reply:
I am still thinking it's necessary to have some tests to verify the correctness of the type definition. I can refactor these tests after we have finish the I/O part (I think it's the next PR).

Reply:
hmm....ok. If we use the pipeline.Builder() and define the pipeline reader API, I expect the tests to be refactored completely.

Reply:
I will work on pipeline IO once this one has been merged

Reply:
Ack.